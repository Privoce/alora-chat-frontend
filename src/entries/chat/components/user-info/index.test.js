import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import {
	UserInfoComponent
} from 'entries/chat/components';

describe('<UserInfoComponent />', () => {
	const modelI = {
		profile: {
			label: 'AM',
			width: 40,
			height: 40,
			backgroundColor: 'red',
			color: 'white',
			labelFontSize: 12
		},
		title: {
			text: 'Amendowins',
			fontSize: 13,
			maxWidth: 100,
			margin: '0px 0px 0px 14px'
		},
		desc: {
			text: 'Hey there! how are you',
			fontSize: 13,
			maxWidth: 100,
			margin: '0px 0px 0px 14px'
		},
		rightLabel: 'right label',
		wrapperStyle: {
			width: '100%',
			justifyContent: 'space-between'
		}
	};

	const getWrapper = (model) => {
		return shallow((
			<UserInfoComponent {...model} />
		));
	};

	it('renders correctly', () => {
		const wrapper = getWrapper(modelI);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
