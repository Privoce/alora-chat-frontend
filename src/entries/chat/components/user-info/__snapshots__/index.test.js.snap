// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfoComponent /> renders correctly 1`] = `
<div
  className="user-info-wrapper"
  style={
    Object {
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <div
    className="user-info-container"
  >
    <ProfilePictureComponent
      backgroundColor="red"
      color="white"
      height={40}
      label="AM"
      labelFontSize={12}
      width={40}
    />
    <div
      className="labels-container"
    >
      <LabelComponent
        breakWord={true}
        dark={true}
        fontSemiBold={true}
        fontSize={13}
        margin="0px 0px 0px 14px"
        maxWidth={100}
        text="Amendowins"
      />
      <LabelComponent
        breakWord={true}
        dark={true}
        fontRegular={true}
        fontSize={13}
        margin="0px 0px 0px 14px"
        maxWidth={100}
        text="Hey there! how are you"
      />
    </div>
  </div>
  <div
    className="info-container"
  >
    <LabelComponent
      alignRight={true}
      fontRegular={true}
      fontSize={13}
      margin="0px 0px 5px 0px"
      text="right label"
      width={100}
    />
    <div
      className="footer-container"
    />
  </div>
</div>
`;
